### PR TITLE
[iOS] Update Mokoresource to use with iOS

### PR DESCRIPTION
**Pull request by kittinunf**
## Issue
- close #119 and #259

## Overview (Required)
- In the previous PRs #441 #450 #462 #482, we have migrated all the string usage for Android and now we already using it on the Android side without much of problem.

- However, this is not the best outcome yet. In order to fully utilize the moko-resource, we should share the string as a single source of truth for iOS as well.

### Changes that I have introduced in this PR

I have migrated the usage of the string resource together with the Android side. The usage on the iOS is the following;

```swift
StringsKt.shared.[string_name].desc().localized()
```

e.g.
```Swift
StringsKt.shared.about_description.desc().localized()
```

Comparing to Android side

```kotlin
stringResource(Strings.about_description)
```

You can check all of the strings that have been declared on the `strings.xml` [here](https://github.com/DroidKaigi/conference-app-2022/blob/main/core-model/src/commonMain/resources/MR/base/strings.xml)

Last but very important information for using the moko-resource on the iOS.

After build, in order to load the bundle correctly, we need to run this gradle task

```bash
"$SRCROOT/../../gradlew" -p "$SRCROOT/../../" :appioscombined:copyFrameworkResourcesToApp \
    -Pmoko.resources.PLATFORM_NAME=$PLATFORM_NAME \
    -Pmoko.resources.CONFIGURATION=$CONFIGURATION \
    -Pmoko.resources.BUILT_PRODUCTS_DIR=$BUILT_PRODUCTS_DIR \
    -Pmoko.resources.CONTENTS_FOLDER_PATH=$CONTENTS_FOLDER_PATH
```

For iOS to be able to run gradlew as a part of the build phase, we have to tell the script where is the java_home location ...

Since the Java_home location is different for each developer, we have to somehow read it from the safe location (easily modifiable / untrack by source control).

Location is here (/local.properties)

Screen Shot 2022-09-18 at 21 50 28

***Important***
From now, in order to run iOS, we need to set the java_home location on the /local.properties file with key `org.gradle.java.home`

e.g. my java_home is the following;

Screen Shot 2022-09-18 at 21 53 23

For @ry-itto, I have tried to use the SPM plugin but with my inexperience on SPM and iOS in general, I couldn't be able to achieve SPM plugin instead of use the run script directly. I would love to use SPM pre-build plugin instead but I couldn't make it work for a few days so I gave up. I have left the trail [here](https://github.com/DroidKaigi/conference-app-2022/pull/578/files#r973717534), just in case you want to explore further.

With this proposal being accepted, [Strings](https://github.com/DroidKaigi/conference-app-2022/tree/main/app-ios/Sources/Strings) module on the iOS will be deprecated and should be removed.

Thank you! and let me know if you have question or problem.. @takahirom @ry-itto 
## Links
-

### CODE REVIEW

File: app-ios/Package.swift [hunk 1]
```diff
@@ -232,15 +233,21 @@ var package = Package(
...
+            checksum: "7586363e24edcf18c2da3ef90f379e9559c1453f48ef5e8fbc0b818fbbc3a045"
         ),
+
         .binaryTarget(
             name: "licenselist",
             url: "https://github.com/touyou/LicenseList/releases/download/0.1.5/licenselist.artifactbundle.zip",
             checksum: "02d1b096c60dd0a4f3ff67a6ec82d801c6a609867fc84aa9ad40d00b42395417"
-        )
+        ),
+//        .binaryTarget(
```
Comment (kittinunf) [inline app-ios/Package.swift:246]: I have left the trail here ... I couldn't make this work with SPM plugin...

File: app-ios/scripts/copy_mokoresources.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+file="../../local.properties"
+
+function prop {
+    grep "${1}" ${file} | cut -d'=' -f2
+}
+
+if [ "$CI" = true ] ; then
+    export PATH="/usr/local/opt/openjdk/bin:$PATH"
+else
+    export JAVA_HOME=$(prop 'org.gradle.java.home')
```
Comment (kittinunf) [inline app-ios/scripts/copy_mokoresources.sh:14]: I am reading java_home from `local.properties` here with key "org.gradle.java.home".

File: app-ios/Package.swift [hunk 3]
```diff
@@ -243,15 +244,21 @@ var package = Package(
             checksum: "9c255e797260054296f9e4e4cd7e1339a15093d75f7c4227b9568d63edddba50"
         ),
         .binaryTarget(
-          name: "swiftgen",
-          url: "https://github.com/SwiftGen/SwiftGen/releases/download/6.6.2/swiftgen-6.6.2.artifactbundle.zip",
-          checksum: "7586363e24edcf18c2da3ef90f379e9559c1453f48ef5e8fbc0b818fbbc3a045"
+            name: "swiftgen",
+            url: "https://github.com/SwiftGen/SwiftGen/releases/download/6.6.2/swiftgen-6.6.2.artifactbundle.zip",
+            checksum: "7586363e24edcf18c2da3ef90f379e9559c1453f48ef5e8fbc0b818fbbc3a045"
```
Comment (ry-itto) [inline app-ios/Package.swift:249]: Please revert this 🙏🏼

File: app-ios/DroidKaigi2022/DroidKaigi2022.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -166,6 +168,27 @@
...
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "./../scripts/copy_mokoresources.sh\n";
+		};
+/* End PBXShellScriptBuildPhase section */
+
```
Comment (ry-itto) [inline app-ios/DroidKaigi2022/DroidKaigi2022.xcodeproj/project.pbxproj:191]: I'm fine with this for now, but I'll fix it later 🙏🏼

**Review by ry-itto — COMMENTED** [review]
> For @ry-itto, I have tried to use the SPM plugin but with my inexperience on SPM and iOS in general, I couldn't be able to achieve SPM plugin instead of use the run script directly. I would love to use SPM pre-build plugin instead but I couldn't make it work for a few days so I gave up. I have left the trail [here](https://github.com/DroidKaigi/conference-app-2022/pull/578/files#r973717534), just in case you want to explore further.

Okey! I'll try this later 👍🏼

**Comment by ry-itto** [comment]
A merged PR https://github.com/DroidKaigi/conference-app-2022/pull/615 has String resource change. 
Please fix conflicts 🙇🏼

File: app-ios/DroidKaigi2022/DroidKaigi2022.xcodeproj/project.pbxproj (hunk 4 above)
Comment (kittinunf) [reply app-ios/DroidKaigi2022/DroidKaigi2022.xcodeproj/project.pbxproj:191]: Sure! Thanks!

File: app-ios/Package.swift (hunk 3 above)
Comment (kittinunf) [reply app-ios/Package.swift:249]: Just fyi, other sections have 4-width indention though. Only the Swiftgen has 2 ...

Comment (ry-itto) [reply app-ios/Package.swift:249]: I see. I understand and I missed...
Thank you..!

**Comment by kittinunf** [comment]
Sure let me work on the conflict tonight then.

**Comment by kittinunf** [comment]
@takahirom with the introduction of project folder structure, it makes the iOS CI choked 😢. (In my humble opinion, if you ask me, I prefer a flat structure rather than this folder setup.)

However, I would love to give this a try. I will try to see whether I can mess around with the iOS bundle creation to solve the issue.

File: app-ios/DroidKaigi2022/iOS/Info.plist [hunk 5]
```diff
@@ -4,5 +4,10 @@
 <dict>
 	<key>ITSAppUsesNonExemptEncryption</key>
 	<false/>
+    <key>CFBundleLocalizations</key>
```
Comment (kittinunf) [inline app-ios/DroidKaigi2022/iOS/Info.plist:7]: This is required from the moko standpoint. It needs one place to read all of the supported languages.

**Comment by takahirom** [comment]
Sorry, I didn't notice that 🙇

**Comment by kittinunf** [comment]
Conflicts fixed! Still finding a good to get around the upload task though 😓. Please wait a bit.

**Comment by kittinunf** [comment]
I have resolved it with using zip instead. Please take a look again. 

For next PR, I will clean up `Strings` for iOS and add some README.md

**Review by ry-itto — APPROVED** [review]
Looks good for me!

> For next PR, I will clean up Strings for iOS and add some README.md

If possible, in order to avoid AppStore Review rejection, I would like you to create an extension that removes the word `Android` from the text for iOS project (I think it should be placed in `Strings` module).

There is some conflicts because https://github.com/DroidKaigi/conference-app-2022/pull/543 merged.

I'm sorry for asking you to fix it so many times. 🙇🏼

**Comment by ry-itto** [comment]
@takahirom 
Please check this PR's Android/Kotlin part 🙏🏼

**Comment by kittinunf** [comment]
> If possible, in order to avoid AppStore Review rejection, I would like you to create an extension that removes the word Android from the text for iOS project (I think it should be placed in Strings module).

Let me check for that in the next PR but I think ideally we should not have the hardcoded string "Android" somewhere. I will help monitor that too

**Comment by kittinunf** [comment]
Sure! Fixed 💪

**Comment by kittinunf** [comment]
@ry-itto It's okay, please ping me when you are merging stuff. I can keep merging stuff, no worries.

**Comment by ry-itto** [comment]
@kittinunf 
I merged that PR 🙏🏼 (This ping is too late..)

**Comment by kittinunf** [comment]
Hopefully CI likes my commit ☺️

**Comment by ry-itto** [comment]
@kittinunf 
CI seems to like your commits!
I'll merge this PR 🙏🏼  Thank you!

@takahirom 
This PR has a risk of conflict, so I'll merge this PR before your review 🙏🏼 
If there is something happened, let's do it with issue!

**Comment by kittinunf** [comment]
Thanks so much! @ry-itto 

Just FYI, iOS build now requires java home to be set on the local.properties file.

**Pull request merged**